src/content/1.7/modules/sample_modules/grid-and-identifiable-object-form-hooks-usage.md

```diff
@@ -0,0 +1,312 @@
+                }
+            }
+        }
+    }
+
+```
+
+This sample demonstrates how to extend sql of the customers grid. From our custom database table `democqrshooksusage_reviewer` we fetch the result of field `is_allowed_for_review`. This name must match the id we added in the grid definition. In order for sorting to work we also add `orderBy` condition and finally, in order
+for filters to work `where` conditions are added if the filter exists in `$searchCriteria->getFilters()`.
+
+#### Result
+
+After completing the steps above by going to customers list you should see new column "allowed for review" added.
+
+{{< figure src="../img/extended_customers_grid.png" title="Allowed for review column added to customers list" >}}
+
+
+### Adding new form field to customer form
+
+#### Modifying customers form builder
+
+In this step we are appending to the customers form new `SwitchType` form field - its one of many form types which already exist in PrestaShop. More information
+about it can be find [here]({{< relref "/1.7/development/components/form/types-reference/_index.md" >}}).
+
+{{% notice note %}}
+**This example has been simplified for practical reasons.** 
+
+You can find full implementation [here](https://github.com/friends-of-prestashop/demo-cqrs-hooks-usage-module) which uses CQRS pattern to get reviewer state. [More about it here]({{< relref "cqrs.md" >}}).
+{{% /notice %}}
+
+```php
+// modules/ps_democqrshooksusage/ps_democqrshooksusage.php
+
+use Symfony\Component\Form\FormBuilderInterface;
+
+public function hookActionCustomerFormBuilderModifier(array $params)
+{
+    /** @var FormBuilderInterface $formBuilder */
+    $formBuilder = $params['form_builder'];
+    $formBuilder->add('is_allowed_for_review', SwitchType::class, [
+        'label' => $this->getTranslator()->trans('Allow reviews', [], 'Modules.Ps_DemoCQRSHooksUsage'),
+        'required' => false,
+    ]);
+    
+    $customerId = $params['id'];
+    
+    $params['data']['is_allowed_for_review'] = $this->getIsAllowedForReview($customerId);
+
+    $formBuilder->setData($params['data']);
+}
+    
+private function getIsAllowedForReview($customerId)
+{
+    // implement your data retrieval logic here
+    
+    return true;
+}
+```
+
+In this sample by using [Symfony form builder](https://symfony.com/doc/current/forms.html) we just added another Form type. To determine if its
+on or off we also need to reset its form data by assigning `is_allowed_for_review` value to `true` or `false`.
+
+#### Result
+
+By completing the steps above newly added switch is now visible in the customers form.
+
+{{< figure src="../img/allow_for_review_switch.png" title="Allowed for review switch added to customers form" >}}
+
+### Extending customers form after create and update actions
+
+In the previous example we have added switch but when we want to save its state (on or off) nothing happens. This is due to we did not used hooks
```

Comment:
```suggestion
In the previous example we have added switch but when we want to save its state (on or off) nothing happens. This is because we have not used the hooks
```